### ISSUE TITLE
Web error: html/template: "script/new" is undefined

### ISSUE THREAD

**Issue by ghost** [tyler-sommer/squircy2#5]
When attempting to add a new script on the configuration page.

**Comment by ghost** [comment]
Tried changing the newScriptAction function `r.HTML(200, "script/new", nil)` to `r.HTML(200, "script/new.html.twig", nil)` but no change.

Also tried changing the function signature to ...

```
func newScriptAction(s *stickHandler) {
    s.HTML(200, "script/new.html.twig", nil)
}
```

... to use a stickHandler instead of render, but no change their either.

Also noticed in the HTML function defined in controller.go, it accepts a status code, but the value is never used, and the response header will always return 200.

**Comment by ghost** [comment]
Figured it out, when using the `go get` method of installing, the program was built and thrown into `$GOPATH/bin` which was being defaulted to in my path when running the command. So I was using the wrong binary.

However, the error is still occurring. Now that my changes are being written to the proper binary, switching to a `s *stickHandler` function signature crashes the program. Renaming from `script/new` to `script/new.html.twig` returns the same undefined error.

**Comment by tyler-sommer** [comment]
This turned out to be an issue in stick as well! After you changed the parameter to `s *stickHandler`, stick was trying to render the page but choking as you noticed, this was the cause: https://github.com/tyler-sommer/stick/commit/d94e78fac65645582ddd264ef176be78f51ff05a

I've fixed stick and updated squircy2 to use the stick handler for rendering the new script template.  Running `go get -u github.com/tyler-sommer/squircy2` should update both packages. Let me know if you have any other issues! Cheers :)

**Comment by ghost** [comment]
That did the trick! Cool beans.

Quick sort of related question. When I make changes to the templates, they do not appear to reflect these changes when running the application again.

Do I need to do anything else outside of editing the templates in `/views`? I was hacking squircy to have TLS support by adding the option in the config model, as well as changing the template for `manage/edit.html.twig` to have a checkbox form field. But running squircy, the template stays stagnant. I tried clearing cache and such, no bueno. I am fairly certain I am using the right binary, as I deleted the one in `$GOPATH/bin` that keeps getting made then I `go get` and I'm launching the binary directly by absolute path. One thing I've noticed, there's no `data` directory in my project directory, so something is leading me to believe that I'm homehow not running the binary from the right location but I've gone to some lengths to try and ensure that I am.

Like, it is somehow saving my settings but there is no `data` directory where the little embedded database was being written to in the working directory where I'm running the binary from. Like, I went so far as to rename the binary just to be sure I was not getting stomped by my path. Is there some other location where it is trying to save things? If it's not pulling the templates from that folder, then where?

**Comment by ghost** [comment]
Ok, figured that out too. It was using the hard path to `github.com/tyler-sommer/squircy2`, but I was working on my own fork in a different directory. So the paths seem to be hardcoded rather than relative to the running directory.

https://github.com/tyler-sommer/squircy2/blob/master/squircy/config/config.go#L8

That should probably be changed to be a runtime value or something.

EDIT:

Addressed by Pull #8

**Comment by tyler-sommer** [comment]
So in merging your PR #8, I [added a CLI flag for `root-path`](https://github.com/tyler-sommer/squircy2/blob/943ec24e033f41511e01d495a04f8b24b4651182/main.go#L9), so now you can run:

``` bash
squircy2 -root-path /some/path
```

This allows an arbitrary root path, but I still am not satisfied with the solution.

Basically I see two usecases:
- **Users who just want to install squircy2 and use it.**
  This kind of installation should default to use `$GOPATH/src/github.com/tyler-sommer/squircy2` as the root path.
- **Users who want to fork squircy2 and hack on it.**
  This user probably doesn't want to have to specify `-root-path` every time they run squircy2, so there should be some mechanism to customize this field. One option I see here is to just change the `basePkg` constant defined in `squircy/config/config.go` to match the fork's name. Then the onus is on me to be sure to not accidentally merge someone else's value if they open a PR.

I'm curious though, what is your development workflow like? Are you in your fork's root directory (so `$GOPATH/src/github.com/Term1nal/squircy2`) and running `go build && ./squircy2` or maybe `go run main.go`?

**Comment by ghost** [comment]
Yeah, my workflow is that I work on my own fork under `github.com/term1nal/squircy2` so I can merge them right back up to my github and do PRs through that. I have to rename my imports and such. What was happening, is that when i would compile and run under my fork, it would go to your repo directory and use the templates from there. So any changes I made in my fork's templates were being ignored. I found that you were hard coding the root. Which is curious, why would you do that instead of how I did it in my PR by finding out the root directory at runtime? Is there a reason for doing it the way you were?

Also, I use `go build` not `go run` (as I believe that go build is the preferred way to do it so I'm told).

**Comment by tyler-sommer** [comment]
Yep that code is specifically designed to allow a user to run squircy2 without any further configuration. Basically what happens is `go get` puts all of the squircy2 source code (including views) into `$GOPATH/src/github.com/tyler-sommer/squircy` and then puts the binary at `$GOPATH/bin/squircy2`. 

Using the current working directory means that the user must install the views themselves into their current working directory for that usecase to function properly.

**Comment by ghost** [comment]
Oh I see, that's assuming they run from `$GOPATH/bin`. Gotcha. Yeah, the command-line option would be pretty nice at the very least, then.

**Comment by tyler-sommer** [comment]
I'm closing this issue as fixed in #10. If you run into any other problem, please open a new issue.

Thanks!
